docs/projects/version-control/creating-release-from-a-build-server-plug-in.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Creating releases from a build server plugin on a version-controlled project
+description: Examples of how to ensure that the right branch is used to create the release when using a build server plugin.
+position: 45 
+---
+
+!include <build-server-plugin-version-control-fields>
+
+Below are some examples of how to auto populate these fields using build parameters for common build servers.
+
+## Azure DevOps
+ 
+Azure DevOps stores information differently based on if the build is triggered by a branch push versus via a Pull Request (PR). You need to use a conditional statement to select the right variable. Here's a suggested method to do so.
+
+For *Git Reference*, use
+```
+$[coalesce(variables['system.pullRequest.sourceBranch'], variables['build.sourceBranch'])]
+```
+
+For *Git Commit*, use
+```
+$[coalesce(variables['system.pullRequest.sourceCommitId'], variables['build.sourceVersion'])]
+```
+
+**Note:** this approach doesn't populate the commit details for manually triggered runs. It is recommended that you provide the values for both branch and commit in this case.
+
+---
+
+## Github Actions
+
+Github Actions provides different event data based on if the build is triggered by a branch push versus via a Pull Request (PR). You need to use a conditional statement to select the right variable. Here's a suggested method to do so.
+
+For *Git Reference*, use
+```
+${{ github.head_ref || github.ref }}
+```
+
+For *Git Commit*, use
+```
+${{ github.event.push.after || github.event.pull_request.head.sha }}
+```
+
+**Note:** this approach doesn't populate the commit details for manually triggered runs. It is recommended that you provide the values for both branch and commit in this case.
+
+---
+
+## TeamCity
+
+TeamCity provides different data based on if the build is triggered by a branch push versus via a Pull Request (PR). There is no easy way to automate selecting the right one. Based on your process, we suggest using one of the following options. 
+
+### Git Reference
+
+If the build is triggered by a branch push, use 
+```
+%teamcity.build.branch%
+```
+If the build is triggered via a PR, use 
+```
+%teamcity.pullRequest.source.branch%
+```
+
+
+### Git Commit
+
+This is the same for both scenarios.
+```
+%build.vcs.number%
+```
+
+### Notes
+If you have multiple VCS sources used in a build, remember to add the fully qualified parameter that includes the VCS Root ID. For example, ```teamcity.build.branch.<VCS_Root_ID>``` and  ```build.vcs.number.<VCS_Root_ID>```
+
+For a build that is the result of a chained step, you may need to reference a parameter from the original build dependency. It will look something like this for a *Git Commit* - ```dep.<Dependant_Build_ID>.build.vcs.number``` or ```dep.<Dependant_Build_ID>.build.vcs.number.<VCS_Root_Id>```.
```

Comment:
```suggestion
For a build resulting from a chained step, you may need to reference a parameter from the original build dependency. It will look something like this for a *Git Commit* - `dep.<Dependant_Build_ID>.build.vcs.number` or `dep.<Dependant_Build_ID>.build.vcs.number.<VCS_Root_Id>`.
```